Improved installation process

chrisozenne opened the pull request:
- CouchDBSample's **main.swift** now creates a connection on port **5984**, the default port used by CouchDB upon install.
- Changed the name of the default database which Kitura tries to connect to: `kitura_test_db`. The README.md is also updated to mention that a database with this name needs to be created.
- The README.md has updated and more thorough installation instructions.
- Improved error logging with HeliumLogger.

chrisozenne committed:
CouchDBSample's main.swift now creates a connection on port 5984, the default port used by CouchDB upon install.
Changed the name of the default database which Kitura tries to connect to: `kitura_test_db`. The README.md is also updated to mention that a database with this name needs to be created.
The README.md has updated and more thorough installation instructions.

chrisozenne committed:
Using HeliumLogger instead of print.

chrisozenne committed:
Adjusted logged errors.

chrisozenne committed:
`ConnectionProperties.swift`:
- Changed from `class` to `struct`
- All properties are non-optional, reducing the need to check for nil.
- `hostName` is now just `host`.
- `userName` is now `username`.
- `HTTPProtocol` is now a computed property.
- `URL` is now a computed property. (and now contains an @ sign between the password and host)
- `toString()` has been replaced with a computed `description` property conforming to the `CustomStringConvertible` protocol.

`CouchDBUtils.swift`:
- Removed some nil-checking code for properties which are no longer optional.
- Updated code to property name changes in other files.

`main.swift`:
- Updated code to property name changes in other files.
- Instead of creating a JSON string literal, then converting it into data and then into a JSON (SwiftyJSON) object, we're now just simply creating a SwiftyJSON object literal and it's much easier to read.
- Added a few // MARK: lines for organization.
- Reorganized the document functions into CRUD order, create, read, update then delete. (With a // MARK: for each)
- In general, formatting matches that of default Xcode formatting (4 spaces instead of 2 for indenting).

chrisozenne committed:
Merge pull request #1 from chrisozenne/feature/cleanup

Cleaned up a few things.

dfirsht commented:
@shmuelk Please review this so it can be merged.

shmuelk commented:
The change to ConnectionProperties, making userid and password required fields with the changes that followed, will cause the API to fail on a server that doesn't have security enabled.

Please change that part back to the way it was.

The rest is very good.
Thanks for your efforts in improving our CouchDB APIs

chrisozenne commented:
@shmuelk Can you point me to where this failure is happening so that I can better understand the issue?

shmuelk commented:
The URL created by:

 // CouchDB URL
 var URL: String {
        return "\(HTTPProtocol)://\(username):\(password)@\(host):\(port)"
 }

Will only work iff there actually is a userid and password. If I am running a local server, I may not want to activate security, yet you are forcing it. The old code build the URL differently depending on whether or not there was a userid and password.

chrisozenne committed:
`username` and `password` for ConnectionProperties are now optional strings once again.
Instead: a warning will be logged if a ConnectionProperties instance is initialized without supplying a username and password.
No longer parsing runtime arguments for connection properties and no longer requiring a username and password among those arguments.

chrisozenne commented:
@shmuelk Please review latest changes. Thanks!

chrisozenne committed:
Updated README contents to match recent changes.

shmuelk commented:
A couple of questions on the latest changes to: CouchDBSample:

  1. Why did you remove the userid and password parameters? They should be made optional.
  2. Why did you hard code the host instead of getting it as a parameter?

I think the number of parameters should be either one or three (i.e. always with a host and with or without a userid and password pair).

chrisozenne commented:
From this comment:

`// Parse runtime args... this is just an interim solution`

It appeared that the previous runtime arguments were just a temporary solution. There was no mention of what they were a temporary solution _to_, so if there is a strong reason for accepting runtime arguments then please let me know and I can change that part back. 

As an outsider, it's not 100% clear to me what the blueprint is, so some more documentation or comments would be very much appreciated so that I don't accidentally break some un-documented requirement.

Thanks.

chrisozenne commented:
@shmuelk can you help me get this merged?

shmuelk commented:
Merged into develop in the commit https://github.com/IBM-Swift/Kitura-CouchDB/commit/6c1725112bcdd008e0d747df47d859fe3e1c52fc
